src/app/screens/Receive/index.tsx

```diff
@@ -44,7 +44,7 @@ function Receive() {
   const [invoice, setInvoice] = useState<{
     paymentRequest: string;
     rHash: string;
-  }>();
+  } | null>();
```

Comment:
as this is a change in the type - can the code handle null in the `invoice` variable? or is somewhere some `invoice.paymentRequest` or something like that in the code?

Reply:
Let me see into it.

Reply:
Nope, checked it. We always check if `invoice` is null or not before proceeding!

Reply:
There are 3 places where we use invoice.PaymentRequest
![image](https://user-images.githubusercontent.com/74018438/218174379-9170c4d9-95e9-4155-93fb-26e28e10dcc5.png)
Would this create an issue?

Reply:
I guess `confirm` can only be called if the invoice is set? - that's how I understand @im-adithya 👍

Reply:
Where are we checking if the invoice is set?

Reply:
> I guess confirm can only be called if the invoice is set?

Ah! No, we enable "Confirm" after we have the formData (i.e. amount and description/memo) and then set the invoice using that, invoice is the result with the QR information here.

```
  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
    event.preventDefault();
    createInvoice();
  }

Reply:
> Would this create an issue?

Nope, that is unrelated to this change.

Reply:
Well then ig null won't be an issue here.

Reply:
> Where are we checking if the invoice is set?

Because the QR you see is coming from the invoice (response):
```
<QRCode value={invoice.paymentRequest.toUpperCase()} level="M" />
```
And to show that, you need to ensure it exists :)

Reply:
> > I guess confirm can only be called if the invoice is set?
> 
> Ah! No, we enable "Confirm" after we have the formData (i.e. amount and description/memo) and then set the invoice using that, invoice is the result with the QR information here.
> 
> ```
>   function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
>     event.preventDefault();
>     createInvoice();
>   }
> ```

So, we generate invoice using formData everytime. So it will always overwrite null?

Reply:
You have to see a new QR each time, so yeah, it should!

Reply:
> You have to see a new QR each time, so yeah, it should!

Well then null won't be an issue!